pkg/controller/machineset/machineset_controller.go

```diff
@@ -230,12 +245,188 @@ func (c *Controller) processNextWorkItem() bool {
+	}
+	c.logger.WithField("machineset", name).Debugf("JDIAZ machinelist: %v", allMachines)
+
+	// get all machines which are children of this machineset
+	var filteredMachines []*clusteroperator.Machine
+	for _, machine := range allMachines {
+		if machine.DeletionTimestamp != nil {
+			continue
+		}
+		controllerRef := metav1.GetControllerOf(machine)
+		if controllerRef == nil {
+			c.logger.WithField("machineset", name).Debugf("JDIAZ no controller ref for machine %v", machine)
+			continue
+		}
+		if ms.UID != controllerRef.UID {
+			//c.logger.WithField("machineset", name).Debug("JDIAZ the kid is not my son %v", machine)
+			continue
+		}
+		filteredMachines = append(filteredMachines, machine)
+	}
+	//c.logger.WithField("machineset", name).Debugf("JDIAZ children machines: %v", filteredMachines)
+
+	var manageMachinesErr error
+
+	//if machineSetNeedsSync && ms.DeletionTimestamp == nil {
+	if true && ms.DeletionTimestamp == nil {
+		manageMachinesErr = c.manageMachines(filteredMachines, ms)
+	}
+
+	if manageMachinesErr != nil {
+		return manageMachinesErr
+	}
+
+	return nil
+}
+
+func (c *Controller) manageMachines(machines []*clusteroperator.Machine, ms *clusteroperator.MachineSet) error {
+	/*machineSetKey, err := controller.KeyFunc(ms)
+	if err != nil {
+		// FIXME
+		return nil
+	}*/
+
+	machinePrefix := ms.ObjectMeta.Name + "-"
+	//c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine prefix: %v", machinePrefix)
+
+	machineSetMachines := []*clusteroperator.Machine{}
+	//machinesToCreate := []*clusteroperator.Machine{}
+	machinesToDelete := []*clusteroperator.Machine{}
+	//machinesToUpdate := []*clusteroperator.Machine{}
+
+	deleteCount := int(0)
+	createCount := int(0)
+
+
+	// caculate how many to create/delete
+	if len(machines) > ms.Spec.Size {
+		deleteCount = len(machines) - ms.Spec.Size
+	}
+	if len(machines) < ms.Spec.Size {
+		createCount = ms.Spec.Size - len(machines)
+	}
+
+	for _, machine := range machines {
+		if ! strings.HasPrefix(machine.ObjectMeta.Name, machinePrefix) {
+			// poorly-named
+			c.logger.WithField("machineset", ms.ObjectMeta.Name).Infof("JDIAZ child machine without expected prefix %v", machine.ObjectMeta.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+
+		if currentMachineNeedsUpdate(machine, ms) {
+			c.logger.WithField("machineset", ms.ObjectMeta.Name).Errorf("JDIAZ child machine needs update %v", machine.ObjectMeta.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+
+		machineSetMachines = append(machineSetMachines, machine)
+
+	}
+
+	// Sync machines
+	machineConfig := buildNewMachine(ms, machinePrefix)
+	c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine template: %+v", machineConfig)
+
+	for i := 0 ; i < createCount ; i++ {
+		_, err := c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Create(machineConfig)
```

Comment:
You are going to want to make these calls in a goroutine so that they can be performed in parallel, but that can be done later.